lib/wraft_doc/search/typesense_server.ex

```
@@ -0,0 +1,93 @@
+defmodule WraftDoc.Search.TypesenseServer do
+  @moduledoc """
+  GenServer to handle Typesense operations for collections and documents.
+  """
+  use GenServer
+  require Logger
+  alias WraftDoc.Search.Typesense
+
+  def start_link(opts \\ []) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    GenServer.start_link(__MODULE__, opts, name: name)
+  end
+
+  def create_collection(schema) do
+    GenServer.call(__MODULE__, {:create_collection, schema}, :timer.seconds(30))
+  end
+
+  def create_document(document) do
+    GenServer.call(__MODULE__, {:create_document, document})
+  end
+
+  def get_document(id, collection_name) when is_binary(id) and is_binary(collection_name) do
+    GenServer.call(__MODULE__, {:get_document, id, collection_name})
+  end
+
+  def update_document(document) do
+    GenServer.call(__MODULE__, {:update_document, document})
+  end
+
+  def delete_document(id, collection_name) when is_binary(id) and is_binary(collection_name) do
+    GenServer.call(__MODULE__, {:delete_document, id, collection_name})
+  end
+
+  @impl true
+  def init(state) do
+    Logger.info("Starting TypesenseServer...")
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_call({:create_collection, schema}, _from, state) do
+    Logger.debug("Creating collection with schema: #{inspect(schema)}")
+    Typesense.create_collection(schema)
```

What if this fails? We should return a failure message in that case right?